src/client/runner/automation/util.js

```diff
@@ -11,27 +13,102 @@ var positionUtils   = testCafeCore.positionUtils;
 var styleUtils      = testCafeCore.styleUtils;
 
 
-function focusElementByLabel (label, callback) {
-    var doc              = domUtils.findDocument(label);
-    var focusableElement = doc.getElementById(label.getAttribute('for'));
+function setCaretPosition (element, caretPos) {
+    var isTextEditable    = domUtils.isTextEditableElement(element);
+    var isContentEditable = domUtils.isContentEditableElement(element);
 
-    if (focusableElement && domUtils.getActiveElement() !== focusableElement)
-        focusBlurSandbox.focus(focusableElement, callback, false, true);
-    else if (callback)
-        callback();
+    if (isTextEditable || isContentEditable) {
+        if (isContentEditable && isNaN(parseInt(caretPos, 10)))
+            textSelection.setCursorToLastVisiblePosition(element);
+        else {
+            var position = isNaN(parseInt(caretPos, 10)) ? element.value.length : caretPos;
+
+            textSelection.select(element, position, position);
+        }
+    }
+    else {
+        // NOTE: if focus is called for not contentEditable element (like 'img' or 'button') inside
+        // contentEditable parent we should try to set right window selection. Generally we can't
+        // set right window selection object because after selection setup window.getSelection method
+        // returns  a different object depending on the browser.
```

Comment:
```
        // NOTE: if focus is called for a non-contentEditable element (like 'img' or 'button') inside
        // a contentEditable parent, we should try to set the right window selection. Generally, we can't
        // set the right window selection object because after the selection setup, the window.getSelection method
        // returns a different object depending on the browser.
```